### PR TITLE
Added wavelength, transmission, and length of data number.

### PR DESCRIPTION
**Pull request by joyvelasquez**

### CODE REVIEW

File: xrtpy/response/channel.py [hunk 1]
```diff
@@ -1,48 +1,553 @@
...
-    def __init__(self, name):
+    def __init__(self,name):
         if name in _channel_name_to_index_mapping:
             self._channel_index = _channel_name_to_index_mapping[name]
             self._channel_data = _genx_file[self._channel_index]
         else:
             raise ValueError(f"{name} is not a valid channel.")
 
+    def __str__(self):
+        """Human reable printout"""
```
Comment (namurphy) [inline xrtpy/response/channel.py:47]:
```suggestion
        """Human readable printout"""
```

File: xrtpy/response/channel.py [hunk 2]
```diff
@@ -1,48 +1,553 @@
...
+        return f"XRT Channel for {self.name}"
+    
+    def __repr__(self):
+        """Code Represenation"""
```
Comment (namurphy) [inline xrtpy/response/channel.py:51]:
```suggestion
        """Code representation"""
```

File: xrtpy/response/channel.py [hunk 3]
```diff
@@ -1,48 +1,553 @@
...
+        return f"Channel({repr(self.name)})"
+
     @property
     def name(self):
```
Comment (namurphy) [inline xrtpy/response/channel.py:55]:
```suggestion
    def name(self) -> str:
```

File: xrtpy/response/channel.py [hunk 4]
```diff
@@ -1,48 +1,553 @@
...
+
+    @property
+    def number_of_wavelengths(self):
+        """
+        Data number length.
+        """
+        return self._channel_data['LENGTH']
+       
+    @property
+    def observatory(self):
```
Comment (namurphy) [inline xrtpy/response/channel.py:84]:
```suggestion
    def observatory(self) -> str:
```

File: xrtpy/response/channel.py [hunk 5]
```diff
@@ -1,48 +1,553 @@
...
+        """
+        Spacecraft: Hinode
+        """
+        return self._channel_data['OBSERVATORY']
+        
+    @property
+    def instrument(self):
```
Comment (namurphy) [inline xrtpy/response/channel.py:91]:
```suggestion
    def instrument(self) -> str:
```

File: xrtpy/response/channel.py [hunk 6]
```diff
@@ -1,48 +1,553 @@
...
+        """
+        X-Ray Telescope - XRT
+        """
+        return self._channel_data['INSTRUMENT']
+
+
+class geometry:
```
Comment (namurphy) [inline xrtpy/response/channel.py:98]:
```suggestion
class Geometry:
```

File: xrtpy/response/channel.py [hunk 7]
```diff
@@ -1,48 +1,553 @@
...
+
+    @property
+    @u.quantity_input
+    def aperture_area(self)-> u.cm**2:
+        """
+        Hinode/XRT Flight Model geometry aperture area.
+        """
+        return u.Quantity(self._geom_data['APERTURE_AREA'], u.cm**2)
+
+class entrance_filter:
```
Comment (namurphy) [inline xrtpy/response/channel.py:129]:
```suggestion
class EntranceFilter:
```

File: xrtpy/response/channel.py [hunk 8]
```diff
@@ -1,48 +1,553 @@
...
+    @property
+    def material(self):
+        """
+        XRT channel Filter-1 Material
+        """
+        return self._fp_filter1_data['MATERIAL']
+
+
+    @property  
+    def filter1_thickness(self): ############
```
Comment (namurphy) [inline xrtpy/response/channel.py:347]:
```suggestion
    def filter1_thickness(self):
```

File: xrtpy/response/channel.py [hunk 9]
```diff
@@ -1,48 +1,553 @@
...
+    @property
+    def name(self,):
+        """
+        Hinode/XRT Flight Model CCD
+        """
+        return self._ccd_data["LONG_NAME"]
+
+    
+    @property
+    def ev_ore_el(self,): ##############
```
Comment (namurphy) [inline xrtpy/response/channel.py:490]:
```suggestion
    def ev_ore_el(self,):
```

File: xrtpy/response/channel.py [hunk 10]
```diff
@@ -1,48 +1,553 @@
...
+        """
+        XRT EV_PER_EL : Units eV
+        """
+        return self._ccd_data['EV_PER_EL']
+
+    @property
+    #@u.quantity_input
+    def full_well(self,): ##############
```
Comment (namurphy) [inline xrtpy/response/channel.py:498]:
```suggestion
    def full_well(self,):
```

File: xrtpy/response/channel.py [hunk 11]
```diff
@@ -1,48 +1,553 @@
...
+    def gain_r(self):
+        """
+        UNITS el per 12 bit ADU
+        """
+        return self._ccd_data['GAIN_R']
+
+    @property
+    def quantum_efficiency(self):
+        """
+        Quantum Efficiency
```
Comment (namurphy) [inline xrtpy/response/channel.py:521]:
```suggestion
        Quantum efficiency.
```

File: xrtpy/response/channel.py [hunk 12]
```diff
@@ -1,48 +1,574 @@
...
 
 import pkg_resources
 import sunpy.io.special
+import numpy as np
 
-filename = pkg_resources.resource_filename(
+from astropy import units as u
+
+filename = pkg_resources.resource_filename
+(
```
Comment (namurphy) [inline xrtpy/response/channel.py:10]:
```suggestion
filename = pkg_resources.resource_filename(
```

**Comment by joyvelasquez** [comment]
Jointed classes e.g. Mirror 1/2 and Filter 1/2.

File: xrtpy/response/channel.py [hunk 13]
```diff
@@ -1,48 +1,449 @@
...
+        """
+        Hinode/XRT flight model Geometry.
+        """
+        return self._geom_data["LONG_NAME"]
+
+    @property
+    @u.quantity_input
+    def focal_len(self,)-> u.cm:  
+        """
+        Hinode/XRT flight model geometry focual length.
```
Comment (namurphy) [inline xrtpy/response/channel.py:55]:
```suggestion
        Hinode/XRT flight model geometry focal length.
```

File: xrtpy/response/channel.py [hunk 14]
```diff
@@ -1,48 +1,449 @@
...
+    _genx_file = _genx_file
+
+    def __init__(self, index):
+        self._channel_index = index
+        self._en_filter_data = self._genx_file[self._channel_index]["EN_FILTER"]
+
+    @property
+    def entrancefilter_name(self,) -> str:
+        """
+        Entrance filters name.
```
Comment (namurphy) [inline xrtpy/response/channel.py:77]:
```suggestion
        Entrance filter name.
```

File: xrtpy/response/tests/test_channel.py [hunk 15]
```diff
@@ -23,3 +23,148 @@
...
+    ("first_property", 1), 
+    ("second_property", "2"), 
+    ("third_property", "three"), # this test should fail
+]
+
+@pytest.mark.parametrize("property, expected_value", properties_and_expected_values)
+def test_accessing_property(property, expected_value):
+    instance = ExampleClass()  # change to name of class being checked
+    actual_value = getattr(instance, property)
+    assert actual_value == expected_value
```
Comment (namurphy) [inline xrtpy/response/tests/test_channel.py:170]:
```suggestion
```

**Review by namurphy — APPROVED** [review]

**Comment by namurphy** [comment]
Looking good!  We can follow up with tests in a different pull request.

**Pull request merged**